src/Compilers/CSharp/Portable/Symbols/AbstractTypeMap.cs

```diff
@@ -232,6 +235,19 @@ private PointerTypeSymbol SubstitutePointerType(PointerTypeSymbol t)
             return new PointerTypeSymbol(pointedAtType);
         }
 
+        private FunctionPointerTypeSymbol SubstituteFunctionPointerType(FunctionPointerTypeSymbol f)
+        {
+            ImmutableArray<TypeWithAnnotations> substitutedParamTypes = SubstituteTypes(f.Signature.ParameterTypesWithAnnotations);
+            var substitutedReturnType = f.Signature.ReturnTypeWithAnnotations.SubstituteType(this);
+
+            if (substitutedParamTypes != f.Signature.ParameterTypesWithAnnotations || !f.Signature.ReturnTypeWithAnnotations.IsSameAs(substitutedReturnType))
```

Comment:
>f.Signature.ParameterTypesWithAnnotations  [](start = 41, length = 42)

This feels fragile. We are taking dependency that ```ParameterTypesWithAnnotations``` returns the same array the second time, however this isn't API's contract, but rather an implementation detail, an optimization. Consider capturing result of first access in a local and avoiding second access.